ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentDeletionScheduler.java

```diff
@@ -15,21 +17,21 @@ public class FragmentDeletionScheduler {
 	private final ApplicationEventPublisher applicationEventPublisher;
 
 	public FragmentDeletionScheduler(FragmentRepository fragmentRepository,
-			ApplicationEventPublisher applicationEventPublisher) {
+	                                 ApplicationEventPublisher applicationEventPublisher) {
 		this.fragmentRepository = fragmentRepository;
 		this.applicationEventPublisher = applicationEventPublisher;
 	}
 
 	@Scheduled(cron = DELETION_CRON_KEY)
 	public void deleteFragments() {
-		fragmentRepository
+		var deletedFragments = fragmentRepository
 				.getDeletionCandidates()
 				.filter(Fragment::isReadyForDeletion)
-				.forEach(readyForDeletionFragment -> {
-					fragmentRepository.removeRelationsPointingToFragmentAndDeleteFragment(readyForDeletionFragment);
-					applicationEventPublisher
-							.publishEvent(new FragmentDeletedEvent(readyForDeletionFragment.getFragmentId()));
-				});
+				.map(Fragment::getFragmentId)
+				.peek(fragmentRepository::removeRelationsPointingToFragmentAndDeleteFragment)
```

Comment:
Sonar will probably tell you as well but using peek outside of debugging is often considered bad practice. Main reasons for this: it creates sides effects, unpredictable when it becomes parallel, execution depends on terminal operations, etc.